Softeq.XToolkit.Common.Tests/Collections/ObservableItemContentRangeCollectionTests/ObservableItemContentRangeCollectionTests.cs

```diff
@@ -0,0 +1,120 @@
+﻿// Developed by Softeq Development Corporation
+// http://www.softeq.com
+
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Collections.Specialized;
+using System.Linq;
+using Softeq.XToolkit.Common.Collections;
+using Softeq.XToolkit.Common.Tests.ObservableObjectTests.Utils;
+using Xunit;
+
+namespace Softeq.XToolkit.Common.Tests.Collections.ObservableItemContentRangeCollectionTests
+{
+    public class ObservableItemContentRangeCollectionTests
+    {
+        private const string TestPropertyValue = "Property value";
+        private const string TestPropertyValue2 = "Property value 2";
+
+        [Theory]
+        [MemberData(
+            nameof(ObservableItemContentRangeCollectionDataProvider.Data),
+            MemberType = typeof(ObservableItemContentRangeCollectionDataProvider))]
+        public void ObservableItemContentRangeCollection_WhenCreatedWithItems_AddsAllItems(
+            IEnumerable<ObservableObject> items)
+        {
+            var count = items?.Count() ?? 0;
+            var collection = new ObservableItemContentRangeCollection<ObservableObject>(items);
+
+            Assert.IsAssignableFrom<ObservableCollection<ObservableObject>>(collection);
+            Assert.Equal(count, collection.Count);
+            if (count > 0)
```

Comment:
Can be split to:
1. Add Null, Empty test
2. Add Items test

Reply:
2 new test cases added